frontend/src/components/Workspace/Visualize/Plot/ImagePlot.tsx

```diff
@@ -554,36 +700,69 @@ const ImagePlotChart = memo(function ImagePlotChart({
   }
 
   const renderActionRoi = () => {
-    if (!roiDataState?.length || outputKey !== "cell_roi") return null
-    if (!isAddRoi) {
-      return <LinkDiv onClick={addRoi}>Add ROI</LinkDiv>
+    if (!roiFilePath || !roiFilePath.includes(CELL_ROI)) return null
+    if (action) {
+      return (
+        <>
+          {action !== "Add ROI" ? (
```

Comment:
@sanglevinh 
Could you please define these "Add ROI", "Merge ROI" and "Delete ROI" as constants?

Reply:
@itutu-tienday i refactored it, pls check it again